src/System.Runtime.WindowsRuntime/src/System/Threading/Tasks/AsyncInfoToTaskBridge.CoreCLR.cs

```diff
@@ -252,5 +241,140 @@ private void Complete(IAsyncInfo asyncInfo, Func<IAsyncInfo, TResult> getResults
                     Marshal.ReleaseComObject(asyncInfo);
             }
         }  // private void Complete(..)
+
+        private static Type s_asyncCausalityTracerType;
+        private static PropertyInfo s_LoggingOn;
+        private static MethodInfo s_TraceOperationCreationMethodInfo;
+        private static MethodInfo s_TraceOperationCompletionMethodInfo; 
+        private bool IsAsyncCausalityTracerLoggingOn()
```

Comment:
Note that this property is expected to be fast. I do not think that the use of reflection the way you have it here would fly well.